salt/_states/metalk8s_package_manager.py

```diff
@@ -0,0 +1,39 @@
+"""Custom state for managing yum packages and dependencies"""
+
+from __future__ import absolute_import
+
+import logging
+
+from salt.exceptions import CommandExecutionError
+
+
+log = logging.getLogger(__name__)
+
+
+__virtualname__ = "metalk8s_package_manager"
+
+
+def __virtualname__():
+
+    if __grains__['os_family'].lower() == 'redhat':
+        return __virtualname__
+    return (False, "metalk8s_package_manager: no yum based system detected")
+
+
+def install_with_deps(name, version=None, fromrepo=None, **kwargs):
+    """Simple helper to manage package downgrade including dependencies."""
+    if version is None:
+        return __states__["pkg.installed"](
+            name=name, fromrepo=fromrepo, **kwargs
+        )
+
+    dep_list = __salt__['metalk8s_package_manager.list_pkg_deps'](
+        name, version, fromrepo
+    )
+    out = []
+    for pkg, version in dep_list.items():
+        pkg_data = {pkg: version}
+        out.append(pkg_data)
```

Comment:
This for will basically just copy the dep_list ? 
This is useless IMO just give `dep_list` directly to `pkg.installed`

Reply:
You're building a `{pkg: version}` list of dicts with only one key/value pair from a dict with a bunch of keys and values.

Reply:
Oh yes it's true MB